Rubberduck.Parsing/VBA/Parsing/AttributeListener.cs

```diff
@@ -44,98 +44,88 @@ public override void EnterModuleVariableStmt(VBAParser.ModuleVariableStmtContext
 
         public override void EnterSubStmt(VBAParser.SubStmtContext context)
         {
+            var attributeScope = (Identifier.GetName(context.subroutineName()), DeclarationType.Procedure);
+            PushNewScope(attributeScope);
+            _membersAllowingAttributes[attributeScope] = context;
+        }
+
+        private void PushNewScope((string scopeIdentifier, DeclarationType scopeType) attributeScope)
+        {
+            _currentScope = attributeScope;
             _currentScopeAttributes = new Attributes();
-            _currentScope = (Identifier.GetName(context.subroutineName()), DeclarationType.Procedure);
-            _membersAllowingAttributes[_currentScope] = context;
         }
 
         public override void ExitSubStmt(VBAParser.SubStmtContext context)
         {
-            if(_currentScopeAttributes.Any())
+            SaveCurrentScopeAttributes(context);
+            PopScope();
+        }
+
+        private void SaveCurrentScopeAttributes(IAnnotatedContext context)
+        {
+            if (_currentScopeAttributes.Any())
             {
                 _attributes.Add(_currentScope, _currentScopeAttributes);
                 context.AddAttributes(_currentScopeAttributes);
             }
-
-            ResetScope();
         }
 
-        private void ResetScope()
+        private void PopScope()
```

Comment:
Much more readable! A minor nitpick; don't you think the `PushScope` and `PopScope` should be placed together, rather being few methods apart?